lib/components/sprite_stack_component.dart

```diff
@@ -0,0 +1,111 @@
+import 'dart:ui';
+
+import '../sprite.dart';
+import '../spritesheet.dart';
+import 'component.dart';
+
+class SpriteStackComponent extends PositionComponent {
+  SpriteSheet spriteSheet;
+
+  static final _sprites = <String, Map<double, Sprite>>{};
```

Comment:
I don't think this should be `static` right? Making this static means that every instance of SpriteStackComponent will use this map.

Reply:
The reason why this is static at the moment is because it generates a Sprite for the given angle for the given image. If we did that at instance level, every instance would generate a sprite for the given angle for that image.

So if we have two instances that have the same image, and the same angle, they will both generate exactly the same Sprite but twice. Meaning twice the memory and CPU.

But to be honest I am not sure if generating the sprites on the go is the best way to go for us, maybe generate a set of sprites at certain angles, but then those angles are determined by giving a `step` value to the constructor? So that we can generate sprites beforehand instead of on the go? What do you think

Reply:
On a side note, the `_generating` and `_generated` are not static so they lose all meaning if we keep the sprites cache on static level.